src/test/java/org/kiwiproject/dropwizard/util/health/UnknownPropertiesHealthCheckTest.java

```diff
@@ -0,0 +1,56 @@
+package org.kiwiproject.dropwizard.util.health;
+
+import static org.kiwiproject.test.assertj.dropwizard.metrics.HealthCheckResultAssertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.kiwiproject.json.LoggingDeserializationProblemHandler;
+
+import java.util.Set;
+
+@DisplayName("UnknownPropertiesHealthCheck")
+class UnknownPropertiesHealthCheckTest {
+
+    private LoggingDeserializationProblemHandler handler;
+    private UnknownPropertiesHealthCheck healthCheck;
+
+    @BeforeEach
+    void setUp() {
+        handler = mock(LoggingDeserializationProblemHandler.class);
+        healthCheck = new UnknownPropertiesHealthCheck(handler);
+    }
+
+    @Nested
+    class IsHealthy {
+
+        @Test
+        void whenHandlerDoesNotHaveAnyUnknownProperties() {
+            when(handler.getUnknownPropertyCount()).thenReturn(0L);
+
+            assertThat(healthCheck)
+                    .isHealthy()
+                    .hasMessage("No unknown properties detected");
+        }
+    }
+
+    @Nested
+    class IsUnhealthy {
+
+        @Test
+        void whenHandlerHasUnknownProperties() {
+            when(handler.getUnknownPropertyCount()).thenReturn(1L);
+
+            var paths = Set.of("badPath");
+            when(handler.getUnexpectedPaths()).thenReturn(paths);
+
+            assertThat(healthCheck)
+                    .isUnhealthy()
+                    .hasMessage("1 unknown properties detected")
```

Comment:
👻 (boo) on "1 properties"  Don't you want to have proper grammar here? 😜

Reply:
I can put the (s) in the message if you want.  It just plugs the number in

Reply:
Actually you can't since the singular ends with a "y" and thus the plural ends with "ies".

The easiest way would be to just have a special case when the count is one, e.g.

```java

// extract a 'count' var in the health check, then:

.hasMessage(unknownPropertiesMessage(count))

// define this method

private static String unknownPropertiesMessage(int count) {
    checkPositive(count, "expected 1 or higher");
    return count == 1 ? "1 unknown property detected" : count + " unknown properties detected";
}
```